packages/edit-site/src/components/global-styles/shadows-edit-panel.js

```diff
@@ -301,18 +301,20 @@ function ShadowEditor( { shadow, onChange } ) {
 	const shadowParts = useMemo( () => getShadowParts( shadow ), [ shadow ] );
 
 	const onChangeShadowPart = ( index, part ) => {
-		shadowParts[ index ] = part;
-		onChange( shadowParts.join( ', ' ) );
+		const newShadowParts = shadowParts.map( ( p, i ) =>
+			i === index ? part : p
+		);
+		onChange( newShadowParts.join( ', ' ) );
 	};
 
 	const onAddShadowPart = () => {
-		shadowParts.push( defaultShadow );
-		onChange( shadowParts.join( ', ' ) );
+		const newShadowParts = [ ...shadowParts, defaultShadow ];
+		onChange( newShadowParts.join( ', ' ) );
```

Comment:
There's no mutation here, so we can just do it inline, without the extra constant IMHO:

```suggestion
		onChange( [ ...shadowParts, defaultShadow ].join( ', ' ) );
```